fix detection mechanism of NVSHMEM libraries
the file 'libnvshmem_device.bc' is only available on Python wheel 'nvidia-nvshmem-cu12'
see https://docs.nvidia.com/nvshmem/release-notes-install-guide/install-guide/nvshmem-install-proc.html#nvshmem-llvm-bitcode-library-based-hello-world-example
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/torch/distributed/_symmetric_memory/_nvshmem_triton.py	2026-09-16 23:33:56.453998000 +0200
+++ b/torch/distributed/_symmetric_memory/_nvshmem_triton.py	2026-09-16 23:25:09.514958000 +0200
@@ -39,15 +39,18 @@
             return cls.found_device_lib_path
 
         # First, check if the user has specified a custom library path
+        libnvshmem_targets = ("libnvshmem_device.bc", "libnvshmem_device.a")
+
         user_lib_dir = os.environ.get("NVSHMEM_LIB_DIR", None)
         if user_lib_dir is not None:
-            lib_path = os.path.join(user_lib_dir, "libnvshmem_device.bc")
-            if not os.path.exists(lib_path):
-                raise RuntimeError(
-                    f"NVSHMEM device library not found at specified path: {user_lib_dir}"
-                )
-            cls.found_device_lib_path = lib_path
-            return lib_path
+            for libnvshmem in libnvshmem_targets:
+                lib_path = os.path.join(user_lib_dir, libnvshmem)
+                if not os.path.exists(lib_path):
+                    raise RuntimeError(
+                        f"NVSHMEM device library not found at specified path: {user_lib_dir}"
+                    )
+                cls.found_device_lib_path = lib_path
+                return lib_path
 
         # Otherwise, search for the library in the default installation paths
         paths = [
@@ -91,10 +94,11 @@
             pass
 
         for path in paths:
-            device_lib = os.path.join(path, "libnvshmem_device.bc")
-            if os.path.exists(device_lib):
-                cls.found_device_lib_path = device_lib
-                return device_lib
+            for libnvshmem in libnvshmem_targets:
+                device_lib = os.path.join(path, libnvshmem)
+                if os.path.exists(device_lib):
+                    cls.found_device_lib_path = device_lib
+                    return device_lib
 
         raise RuntimeError(f"NVSHMEM device library not found. Searched: {paths}")
 
